Track/adjust for test failure with genno 1.23.0

(Apparently) with genno 1.23.0, we see failures like [this](https://github.com/iiasa/message_ix/actions/runs/7676514776/job/20924167977#step:9:420):
```
>       assert_qty_equal(obs, demand, check_attrs=False)

message_ix/tests/test_report.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

left = Index(['chicago', 'new-york', 'topeka'], dtype='object', name='n')
right = MultiIndex([( 'chicago',),
            ('new-york',),
            (  'topeka',)],
           names=['n'])
obj = 'Series.index'

    def _check_types(left, right, obj: str = "Index") -> None:
        if not exact:
            return
    
>       assert_class_equal(left, right, exact=exact, obj=obj)
E       AssertionError: Series.index are different
E       
E       Series.index classes are different
E       [left]:  Index(['chicago', 'new-york', 'topeka'], dtype='object', name='n')
E       [right]: MultiIndex([( 'chicago',),
E                   ('new-york',),
E                   (  'topeka',)],
E                  names=['n'])
```

This may actually be a regression in `genno` due to some housekeeping in khaeru/genno#115, that adapted AttrSeries.\_\_init__ to address deprecation warnings with pandas 2.2. What appears to have happened is that the code now skips [this block](https://github.com/khaeru/genno/blob/6cedebc53c46cd6df2f8c3bb2a665c96d8f1d4cf/genno/core/attrseries.py#L160-L168) that ensures a pd.MultiIndex. So `obs` in this test, while *functionally* identical to `demand`, has a different index class and thus fails assert_qty_equal().

This fix is probably upstream, this is just to track.

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -22,5 +22,34 @@ updates:
     # Add reviewers:
     reviewers: 
       - "glatterf42" 
+      - "khaeru"
+  - package-ecosystem: "github-actions"
+    # Only update major versions
+    ignore:
+      - dependency-name: "*"
+        update-types:
+        - "version-update:semver-minor"
+        - "version-update:semver-patch"
+    # Below config mirrors the example at
+    # https://github.com/dependabot/dependabot-core/blob/main/.github/dependabot.yml
+    directory: "/"
+    schedule:
+      interval: "weekly"
+      day: "sunday"
+      time: "16:00"
+    groups:
+      all-actions:
+        patterns: [ "*" ]
+     # Add assignees 
+    assignees: 
+      - "glatterf42" 
       - "khaeru" 
+    # Specify labels 
+    labels: 
+      - "dependencies" 
+    # Add reviewers:
+    reviewers: 
+      - "glatterf42" 
+      - "khaeru" 
+  
     

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -26,7 +26,7 @@ jobs:
 
     steps:
     - name: Cache Anaconda installer, conda packages
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: |
           $CONDA/pkgs

--- a/.github/workflows/nightly.yaml
+++ b/.github/workflows/nightly.yaml
@@ -50,7 +50,7 @@ jobs:
       run: python -m pip install --upgrade pip setuptools-scm wheel
 
     - name: Cache GAMS installer and Python packages
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: gams
         key: ubuntu-latest-gams${{ env.GAMS_VERSION }}

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -65,7 +65,7 @@ jobs:
         macos-skip-brew-update: true
 
     - name: Cache GAMS installer and R packages
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: |
           gams
@@ -85,8 +85,6 @@ jobs:
       run: |
         pip install --upgrade "ixmp @ git+https://github.com/iiasa/ixmp.git@main"
         pip install .[tests]
-        # TEMPORARY work around iiasa/message_ix#788
-        pip install "genno<1.23"
 
     - name: Run test suite using pytest
       env:
@@ -101,7 +99,9 @@ jobs:
       shell: bash
 
     - name: Upload test coverage to Codecov.io
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@v4
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
   tutorials:
     strategy:
@@ -141,7 +141,7 @@ jobs:
       id: setup-r
 
     - name: Cache GAMS installer and R packages
-      uses: actions/cache@v3
+      uses: actions/cache@v4
       with:
         path: |
           gams
@@ -185,7 +185,9 @@ jobs:
       shell: bash
 
     - name: Upload test coverage to Codecov.io
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@v4
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
   pre-commit:
     name: Code quality

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -59,11 +59,11 @@ jobs:
         python-version: ${{ matrix.python-version }}
         cache: pip
         cache-dependency-path: "**/pyproject.toml"
-  
+
     - uses: ts-graphviz/setup-graphviz@v2
       with:
         macos-skip-brew-update: true
-      
+
     - name: Cache GAMS installer and R packages
       uses: actions/cache@v4
       with:
@@ -132,7 +132,7 @@ jobs:
       # Use the environment variable set by the setup-python action, above.
       run: echo "RETICULATE_PYTHON=$pythonLocation" >> "$GITHUB_ENV"
       shell: bash
-    
+
     - uses: ts-graphviz/setup-graphviz@v2
       with:
         macos-skip-brew-update: true

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -59,11 +59,11 @@ jobs:
         python-version: ${{ matrix.python-version }}
         cache: pip
         cache-dependency-path: "**/pyproject.toml"
-
-    - uses: ts-graphviz/setup-graphviz@v1.2.0
+  
+    - uses: ts-graphviz/setup-graphviz@v2
       with:
         macos-skip-brew-update: true
-
+      
     - name: Cache GAMS installer and R packages
       uses: actions/cache@v4
       with:
@@ -132,13 +132,15 @@ jobs:
       # Use the environment variable set by the setup-python action, above.
       run: echo "RETICULATE_PYTHON=$pythonLocation" >> "$GITHUB_ENV"
       shell: bash
-
-    - uses: ts-graphviz/setup-graphviz@v1.2.0
+    
+    - uses: ts-graphviz/setup-graphviz@v2
       with:
         macos-skip-brew-update: true
 
     - uses: r-lib/actions/setup-r@v2
       id: setup-r
+      with:
+        windows-path-include-rtools: false
 
     - name: Cache GAMS installer and R packages
       uses: actions/cache@v4

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -130,7 +130,7 @@ jobs:
 
     - name: Set RETICULATE_PYTHON
       # Use the environment variable set by the setup-python action, above.
-      run: echo "RETICULATE_PYTHON=$pythonLocation" >> $GITHUB_ENV
+      run: echo "RETICULATE_PYTHON=$pythonLocation" >> "$GITHUB_ENV"
       shell: bash
 
     - uses: ts-graphviz/setup-graphviz@v1.2.0

--- a/.github/workflows/pytest.yaml
+++ b/.github/workflows/pytest.yaml
@@ -188,8 +188,8 @@ jobs:
 
     - name: Upload test coverage to Codecov.io
       uses: codecov/codecov-action@v4
-      env:
-        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }} # required
 
   pre-commit:
     name: Code quality

